fw/access_log.c

```diff
@@ -413,21 +543,89 @@ do_access_log(TfwHttpResp *resp)
 			  TFW_HTTP_RESP_CUT_BODY_SZ(resp));
 }
 
+static int
+cfg_access_log_set(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	int i;
+	const char *val;
+
+	TFW_CFG_CHECK_VAL_N(>, 0, cs, ce);
+	TFW_CFG_CHECK_NO_ATTRS(cs, ce);
+
+	access_log_type = ACCESS_LOG_OFF;
+
+	TFW_CFG_ENTRY_FOR_EACH_VAL(ce, i, val) {
+		if (strcasecmp(val, "off") == 0)
+			break;
+
+		if (strcasecmp(val, "dmesg") == 0) {
+			access_log_type |= ACCESS_LOG_DMESG;
+		} else if (strcasecmp(val, "mmap") == 0) {
+			access_log_type |= ACCESS_LOG_MMAP;
+		} else {
+			T_ERR_NL("invalid access_log value: '%s'\n", val);
+			return -EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static int
+tfw_access_log_start(void)
+{
+	int cpu;
+
+	if (!(access_log_type & ACCESS_LOG_MMAP))
+		return 0;
+
+	if (mmap_buffer)
```

Comment:
It seems that it is an extra check, because we have the same check in `tfw_mmap_buffer_free`

Reply:
Fixed.